sdk/src/runtime/worker.tsx

```diff
@@ -144,15 +149,29 @@ export const defineApp = <T extends RequestInfo = RequestInfo<any, DefaultAppCon
 
           const [rscPayloadStream1, rscPayloadStream2] = rscPayloadStream.tee();
 
-          const htmlStream = await transformRscToHtmlStream({
-            stream: rscPayloadStream1,
-            Document: rw.Document,
-            requestInfo: requestInfo,
-          });
+          let html: ReadableStream<any>;
+
+          if (rw.ssr) {
+            html = await transformRscToHtmlStream({
+              stream: rscPayloadStream1,
+              Document: rw.Document,
+              requestInfo: requestInfo,
+            });
+          } else {
+            const emptyRscStream = renderToRscStream({
+              node: React.createElement(React.Fragment, null, null),
+              actionResult: undefined,
+              onError,
+            });
+            html = await transformRscToHtmlStream({
+              stream: emptyRscStream,
+              Document: rw.Document,
+              requestInfo: requestInfo,
+            });
+          }
```

Comment:
Converts RSC to HTML or, not.

Reply:
Aaah, ok! So we return just the shell for that page, clever!